src/models/network/NetworkBaseController.js

```diff
@@ -124,28 +108,22 @@ export default class NetworkBaseController {
   }
 
   hasContractChanged(contractAlias) {
-    const contractName = this.packageData.contracts[contractAlias];
-    if (!this.isApplicationContract(contractAlias)) return false;
+    if (!this.packageFile.hasContract(contractAlias)) return false;
     if (!this.isContractDeployed(contractAlias)) return true;
+    const contractName = this.packageFile.contract(contractAlias);
     const contractClass = Contracts.getFromLocal(contractName);
-    const currentBytecode = bytecodeDigest(contractClass.bytecode);
-    const deployedBytecode = this.networkPackage.contracts[contractAlias].bytecodeHash;
-    return currentBytecode !== deployedBytecode;
-  }
-
-  isApplicationContract(contractAlias) {
-    return !!this.packageData.contracts[contractAlias];
+    return this.networkFile.hasSameBytecode(contractAlias, contractClass)
```

Comment:
Shouldn't be the negation (ie contract has changed if bytecode is *not* the same)? More important: why didn't any test catch this?

Reply:
Oh! `hasSameBytecode` was actually checking if both bytecodes were different 😅 
I'll fix that